htdocs/comm/propal/list.php

```diff
@@ -629,8 +629,13 @@
 
 $sql .= ' WHERE p.fk_soc = s.rowid';
 $sql .= ' AND p.entity IN ('.getEntity('propal').')';
-if (empty($user->rights->societe->client->voir) && !$socid) { //restriction
-	$sql .= " AND s.rowid = sc.fk_soc AND sc.fk_user = ".((int) $user->id);
+if (!$user->hasRight('societe', 'client', 'voir') && !$socid) { //restriction
+	$sql .= " AND s.rowid = sc.fk_soc AND (sc.fk_user = ".((int) $user->id);
+	$userschilds = $user->getAllChildIds();
```

Comment:
the getAllChildIds return the hierachy and the permission management on hierarchy is supported only for the modules HR (expense report and holiday).

The link with "sales represnetatives" is planned to have a rewrite (because it is often managed by a left join, when we need a WHERE only).  This is a postponed delay, and i suggest to wait this task is completed (and we will try to include your need) before adding the feature to manage the hierarchy into link with sale representatives...

Reply:
@eldy not return only id tu subordonate

Reply:
Link with sales representative to make filter on sale representative has been rewrite with no join in v19